superset/dashboards/dao.py

```diff
@@ -57,6 +58,29 @@ def get_by_id_or_slug(id_or_slug: str) -> Dashboard:
             raise DashboardNotFoundError()
         return dashboard
 
+    @staticmethod
+    def get_datasets_for_dashboard(id_or_slug: str) -> Dict[str, Any]:
+        query = (
+            db.session.query(Dashboard)
+            .filter(id_or_slug_filter(id_or_slug))
+            .outerjoin(Slice, Dashboard.slices)
+            .outerjoin(Slice.table)
+        )
+        # Apply dashboard base filters
+        query = DashboardFilter("id", SQLAInterface(Dashboard, db.session)).apply(
+            query, None
+        )
+        dashboard = query.one_or_none()
+        if not dashboard:
+            raise DashboardNotFoundError()
```

Comment:
Feels kind of strange that a `DAO` raises a `Command` Exception

Reply:
Oh I never noticed that. It also feels kind of strange to create multiple different exception classes with the exact same meaning. I think it's reasonable to share them.